src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java

```diff
@@ -56,31 +58,27 @@ public class LuceneFieldMapper extends KNNVectorFieldMapper {
         );
 
         this.knnMethod = input.getKnnMethodContext();
-
-        this.fieldType = new FieldType();
-
-        this.fieldType.setTokenized(false);
-        this.fieldType.setIndexOptions(IndexOptions.NONE);
-
         final SpaceType spaceType = this.knnMethod.getSpaceType();
         final VectorSimilarityFunction vectorSimilarityFunction = Optional.ofNullable(
             SPACE_TYPE_TO_VECTOR_SIMILARITY_FUNCTION.get(spaceType)
         ).orElseThrow(() -> new IllegalArgumentException(String.format("Space type [%s] is not supported for Lucene engine", spaceType)));
-
         final int dimension = input.getMappedFieldType().getDimension();
-        if (dimension > MAX_DIMENSION) {
-            throw new IllegalArgumentException(
-                String.format(
-                    "Dimension value cannot be greater than [%s] but got [%s] for vector [%s]",
-                    MAX_DIMENSION,
-                    dimension,
-                    input.getName()
-                )
-            );
+
+        this.fieldType = KnnVectorField.createFieldType(dimension, vectorSimilarityFunction);
+
+        if (this.hasDocValues) {
+            this.docValuesFieldType = buildDocValuesFieldType(this.knnMethod.getKnnEngine());
+        } else {
+            this.docValuesFieldType = null;
         }
```

Comment:
Explicit setting the value to null is not required. 

Also, why we are creating this fieldType here? can we create a single fieldType which tells that this field will be stored, will have doc values.

Reply:
> Explicit setting the value to null is not required.

docValuesFieldType is final so it needs to be explicitly set. 

> Also, why we are creating this fieldType here? can we create a single fieldType which tells that this field will be stored, will have doc values.

When user uses doc values, we are creating 2 Lucene fields for this OpenSearch field. One for binary doc values and another for the k-NN vector values. If we use just the "fieldType" fieldType and set DocValuesType to BinaryDocValues and create KnnVectorField with it, it will fail because KnnVectorField does not support binary doc values ([This method](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/document/Field.java#L421) will eventually be called and return null).

So, we need 2 fieldTypes for 2 fields.